nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/PriorityQueueShufflingAdder.java

```diff
@@ -176,35 +172,18 @@ public void visitEnd() {
         addNext();
         addHasNext();
         addElements();
-        addIndex();
-
+        addIter();
         super.visitEnd();
     }
 
     @Override
     public MethodVisitor visitMethod(int access, String name, String desc,
                                      String signature, String[] exceptions) {
         if ("hasNext".equals(name)) {
-            hasNextProp.access = access;
-            hasNextProp.name = name;
-            hasNextProp.desc = desc;
-            hasNextProp.signature = signature;
-            hasNextProp.exceptions = exceptions;
-
-            MethodVisitor original = super.visitMethod(access, "hasNextOrig", desc, signature, exceptions);
-
-            return original;
+            return super.visitMethod(access, "hasNextOrig", desc, signature, exceptions);
         }
         if ("next".equals(name)) {
```

Comment:
Add an if here to check if this class has already a constructor, i.e., <init> method. If so, we'd need to change the current instrumentation, which just generates an <init> method from scratch, to add the code to the existing <init> methods.